### DISCUSSION TITLE
Not able to get value of Foreign-key model because of `weak proxy`

### DISCUSSION THREAD

**Discussion by harshithjv** [collerek/ormar#418] [General]
I think my issue is similar to what is described in this issuelink, as I'm not able to get attribute of foreign key:
https://github.com/collerek/ormar/issues/118

I get `weakproxy` on foreign-key object (as seen in VS Code debug window): 
```
{
    'id': 2,
    'things`: [
          <weakproxy at 0x7fe4a528db80 to OtherThing at 0x7fe4a528c340>
   ]
}
```

Say, I want to access name of `OtherThing` from the instance of `Thing` model. Then I'd try to access it like this:

```
t1 = Thing.objects.get_or_create(...)
print(t1.other_thing.name)
```

So `t1.other_thing.name` return empty `None` because I can only access. 

Also I tried to manually getting value of foreign-key object when :

```
ot1 = OtherThing.objects.get(id=t1.other_thing.id)
print(ot1.name)
```

I just get `coroutine(...)` object and not able to display proper data via jinja templates on fastapi. Even `asyncio.run(...)` gives out coroutine object.

How can we access the foreign-key model object seamlessly from parent model?

**Comment by collerek** [comment]
In your sample, you do not await any of the calls (that's why you get a coroutine)
It should be:
```python
t1 = await Thing.objects.get_or_create(...)
# (...)
ot1 = await OtherThing.objects.get(id=t1.other_thing.id)
```

**Reply by harshithjv** [reply]
@collerek 

Ha. I missed that while writing the example. Will update. However, even `await` results in `coroutine` object instead of value.

In my model, I did this like this:

```
class OtherThing(ormar.Model):
      ...
      @ormar.property_field
      async def thing_name(self):
             thing = await Thing.objects.get(id=self.thing.id)
             return thing.name
```

I want to use `thing_name` property as `{{ ot.thing_name }}` within jinja template. But I get something like `<coroutine object OtherThing.thing_name at 0x7fb7712528c0>` everytime.

I tried above property method without `async` method I get this message on console: `RuntimeWarning: coroutine 'QuerySet.get' was never awaited  return getattr(obj, attribute)` which is obivious as we need to use only coroutine as per your design. But tried out since `async - await` is new to me.

**Reply by collerek** [reply]
Unfortunately, property_filed is sync only.  Check the answer in #443.

But you should be able to access a chained property like `thing.other_thing.name`.
If you can paste full example I can try to help.
